### PR TITLE
Clarify meta tags documentation

### PR DESCRIPTION
**Pull request by mikeweilgart**

### CODE REVIEW

**Comment by mikeweilgart** [comment]
Ah, somehow I had missed [this tutorial](https://docs.cfengine.com/docs/3.15/examples-tutorials-tags.html) - but the documentation for the meta attribute should still be clear on its own.

File: reference/promise-types.markdown [hunk 1]
```diff
@@ -113,7 +113,30 @@ A "meta" attribute can likewise be added into any body (of any type).
...
+```
+
+The meta tags may be referred to programmatically in various ways, or may be solely for
+human consumption.  Meta tags on vars promises and classes promises are
+particularly suited for programmatic interpretation; meta tags on other
+promise types (or in bodies) are more likely to be intended only for human consumption.
+
+Relevant CFEngine functions are: `classesmatching()`,
+`classmatch()`, `countclassesmatching()`, `getclassmetatags()`,
+`getvariablemetatags()`, `variablesmatching()`, `variablesmatching_as_data()`
```
Comment (vpodzime) [inline reference/promise-types.markdown]: Are these going to be rendered as links to the particular functions' documentation?

File: reference/promise-types.markdown [hunk 2]
```diff
@@ -113,7 +113,30 @@ A "meta" attribute can likewise be added into any body (of any type).
...
+
+Also see "meta promises": there can be promises of type "meta", in addition
```
Comment (vpodzime) [inline reference/promise-types.markdown]: *meta promises* should also be a link here.

File: reference/promise-types.markdown [hunk 3]
```diff
@@ -113,7 +113,30 @@ A "meta" attribute can likewise be added into any body (of any type).
...
+to this attribute with the name "meta" which can be added to a promise of any type.
```
Comment (vpodzime) [inline reference/promise-types.markdown]: *there can be promises of type "meta"....which can be added to promise of any type* -- that's really confusing.

File: reference/promise-types.markdown (hunk 1 above)
Comment (mikeweilgart) [reply reference/promise-types.markdown]: I believe so.  I imitated other parts of the documentation where they are so rendered.

File: reference/promise-types.markdown (hunk 3 above)
Comment (mikeweilgart) [reply reference/promise-types.markdown]: There's a comma for separation.  The attribute named "meta" that's documented here can be added to a promise of any type.  There can also be promises of type "meta."

Comment (mikeweilgart) [reply reference/promise-types.markdown]: I agree that could be worded better.  Maybe:

Also see "meta promises."  We've been talking about the attribute named "meta" which can be added to a promise of any type; there can also be promises of *type* "meta."

File: reference/promise-types.markdown (hunk 2 above)
Comment (mikeweilgart) [reply reference/promise-types.markdown]: I agree; hopefully Nick will know how to do that.  (I know how to do it in markdown, but I've no idea how to make the link work correctly for this documentation repo, without hardcoding the CFEngine version in the URL.)

File: reference/promise-types.markdown (hunk 3 above)
Comment (vpodzime) [reply reference/promise-types.markdown]: What about:
While "meta" **attribute** can be added to a promise of any type, there can also be **promises** of promise type "meta" added to any bundle.

**Comment by nickanderson** [comment]
Nice, thanks for the contribution @mikeweilgart !

I will merge this and handle the followups with backporting and tidying up the note about meta promises.

**Pull request merged**